src/symbol_table.rs

```diff
@@ -61,6 +61,18 @@ impl SymbolTable {
         id
     }
 
+    /// adds `text` to the symbol table and returns the newly assigned [SymbolId].
+    pub(crate) fn add_symbol<A: AsRef<str>>(&mut self, text: A) -> SymbolId {
+        let text = text.as_ref();
+        // Otherwise, intern it and return the new ID.
```

Comment:
I think this comment was copy/pasted from `fn intern` above? Consider having `fn intern` call `add_symbol` so we cut down on the duplicated code.